.github/workflows/checks.yml

```diff
@@ -1,52 +1,50 @@
 name: checks
 
-on:
-  push:
-    branches: [main]
-  pull_request:
-    branches: [main]
+on: [push, pull_request]
 
 jobs:
   formatting:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         python-version: ["3.9"]
-        poetry-version: ["1.3.1"]
+        poetry-version: ["1.5.1"]
     steps:
       - uses: actions/checkout@v2
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.python-version }}
       - name: Set up Poetry ${{ matrix.poetry-version }}
         uses: abatilo/actions-poetry@v2
         with:
           poetry-version: ${{ matrix.poetry-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "poetry"
```

Comment:
Surely you have to set up python before poetry, otherwise what version of python and pip is poetry using? Presumably whichever the system version in the runner image (ubuntu-latest) is, which could break in the future if the versions mismatch enough to cause a problem

Reply:
@dni ?

Reply:
if you set it up differntly the it is not using the cache though, any ideas? or just scrap the cache?